File: src/백도현/week5/5-2/pgs118667_dh.py
from collections import deque
def solution(queue1, queue2):
    answer = 0
    dq1 = deque()
    dq2 = deque()
    sum1 = 0
    sum2 = 0
    for i in range(len(queue1)):
        dq1.appendleft(queue1[i])
        dq2.appendleft(queue2[i])
        sum1 += queue1[i]
        sum2 += queue2[i]
        
    target = (sum1 + sum2) / 2
    if (sum1 + sum2) % 2 != 0:
        return -1
    while True:
        if answer > len(queue1) * 3:
            return -1
        if sum1 == target:
            break
        elif sum1 > target:
            sum1 -= dq1[-1]
            dq2.appendleft(dq1.pop())
        else :
            sum1 += dq2[-1]
            dq1.appendleft(dq2.pop())
        answer += 1
            
    return answer

File: src/백도현/week5/5-2/pgs118667_dh.java
import java.util.*;
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        long sum1 = 0;
        long sum2 = 0;
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();
        for (int i = 0; i < queue1.length; i++) {
            sum1 += queue1[i];
            sum2 += queue2[i];
            q1.add(queue1[i]);
            q2.add(queue2[i]);
        }
        long k = (sum1+sum2) / 2;
        
        if ((sum1+sum2)%2 != 0) return -1;
        
        int i = 0;
        int j = 0;
        while (true) {
            if (answer > queue1.length * 3) {
                return -1;
            }
            if (sum1 == k)
                break;
            else if (sum1 > k) {
                sum1 -= q1.peek();
                q2.add(q1.poll());
            } else {
                sum1 += q2.peek();
                q1.add(q2.poll());
            }
            answer++;
        }
        
        
        return answer;
    }
}